stm32-modules/heater-shaker/include/heater-shaker/errors.hpp

```diff
@@ -1,62 +1,56 @@
 #pragma once
 #include <charconv>
+#include <cstdint>
 
 #include "heater-shaker/utility.hpp"
 
 namespace errors {
 
+enum class MotorErrorOffset : uint8_t {
+    FOC_DURATION = 0,
+    OVER_VOLT = 1,
+    UNDER_VOLT = 2,
+    OVER_TEMP = 3,
+    START_UP = 4,
+    SPEED_FDBK = 5,
+    OVERCURRENT = 6,
+    SW_ERROR = 7,
+};
+
+template <typename IntType>
+auto operator<<(IntType to_shift, MotorErrorOffset offset) -> IntType {
```

Comment:
Can you explain the purpose of shifting the motor error offset?

Reply:
Oh actually it's kind of useful if you look at the tests. It lets you do
```
uint16_t value = 1 << MotorErrorOffset::FOC_DURATION;
```
rather than 
```
uint16_t value = 1 << static_cast<uint8_t>(MotorErrorOffset::FOC_DURATION);
```
everywhere